### PR TITLE
1822 v2 check favorite doesn't refresh timeline

### PR DESCRIPTION
**Pull request by ramtob**

### CODE REVIEW

File: src/app/@ansyn/core/utils/overlays.ts [hunk 1]
```diff
@@ -7,12 +7,11 @@ export function isFullOverlay(overlay: IOverlay): boolean {
 	return Boolean(overlay && overlay.date);
 }
 
-export function buildFilteredOverlays(overlays: IOverlay[], parsedFilters: IFilterModel[], favorites: IOverlay[], showOnlyFavorite: boolean, removedOverlaysIds: string[], removedOverlaysVisibility: boolean): string[] {
-	let parsedOverlays: IOverlay[] = favorites;
-	if (!showOnlyFavorite) {
-		const filteredOverlays = overlays.filter((overlay) => parsedFilters.every(filter => filter.filterFunc(overlay, filter.key)));
-		parsedOverlays = [...parsedOverlays, ...filteredOverlays];
-	}
+export function buildFilteredOverlays(overlays: IOverlay[], parsedFilters: IFilterModel[], removedOverlaysIds: string[], removedOverlaysVisibility: boolean): string[] {
```
Comment (yairtawil) [inline src/app/@ansyn/core/utils/overlays.ts]: very cool 😎

File: src/app/@ansyn/menu-items/filters/services/filters.service.ts [hunk 2]
```diff
@@ -64,7 +64,7 @@ export class FiltersService {
 		const cloneFilters = new Map(filterState.filters);
 		cloneFilters.set(metadataKey, cloneMetadata);
 		const filterModels: IFilterModel[] = this.pluckFilterModels(cloneFilters);
-		const filteredOverlays: string[] = buildFilteredOverlays(Array.from(overlays.values()), filterModels, favoriteOverlays, filterState.facets.showOnlyFavorites, removedOverlaysIds, removedOverlaysVisibility);
+		const filteredOverlays: string[] = buildFilteredOverlays(Array.from(overlays.values()), filterModels, removedOverlaysIds, removedOverlaysVisibility);
```
Comment (yairtawil) [inline src/app/@ansyn/menu-items/filters/services/filters.service.ts]: use MapValueToArray

File: src/app/@ansyn/overlays/effects/overlays.effects.ts [hunk 3]
```diff
@@ -35,12 +32,10 @@ export class OverlaysEffects {
...
 	loadOverlays$: Observable<LoadOverlaysSuccessAction> = this.actions$.pipe<any>(
 		ofType<LoadOverlaysAction>(OverlaysActionTypes.LOAD_OVERLAYS),
-		withLatestFrom(this.store$.select(coreStateSelector)),
-		mergeMap(([action, { favoriteOverlays }]: [LoadOverlaysAction, ICoreState]) => {
+		mergeMap((action: LoadOverlaysAction) => {
 			return this.overlaysService.search(action.payload).pipe(
 				mergeMap((overlays: IOverlaysFetchData) => {
-					const overlaysResult = unionBy(Array.isArray(overlays.data) ? overlays.data : [],
-						favoriteOverlays, o => o.id);
+					const overlaysResult = Array.isArray(overlays.data) ? overlays.data : [];
```
Comment (yairtawil) [inline src/app/@ansyn/overlays/effects/overlays.effects.ts]: thanks

File: src/app/@ansyn/overlays/reducers/overlays.reducer.ts [hunk 4]
```diff
@@ -1,9 +1,9 @@
 import { CoreActionTypes, IOverlay, IOverlaySpecialObject, SetOverlaysCriteriaAction } from '@ansyn/core';
-import { OverlaysService } from '../services/overlays.service';
 import { LoadOverlaysSuccessAction, OverlaysActions, OverlaysActionTypes } from '../actions/overlays.actions';
 import { createFeatureSelector, createSelector, MemoizedSelector } from '@ngrx/store';
 import * as _ from 'lodash';
 import { ExtendMap } from './extendedMap.class';
+import { mapToArray } from '../../core/utils/misc';
```
Comment (yairtawil) [inline src/app/@ansyn/overlays/reducers/overlays.reducer.ts]: public_api

File: src/app/@ansyn/ansyn/app-effects/effects/filters.app.effects.spec.ts [hunk 5]
```diff
@@ -18,15 +18,23 @@ import {
...
-import { LoadOverlaysAction, LoadOverlaysSuccessAction, OverlayReducer, overlaysFeatureKey } from '@ansyn/overlays';
+import {
+	LoadOverlaysAction,
+	LoadOverlaysSuccessAction,
+	OverlayReducer,
+	overlaysFeatureKey,
+	OverlaysService, overlaysStatusMessages, SetFilteredOverlaysAction, SetOverlaysStatusMessage
+} from '@ansyn/overlays';
 import { menuFeatureKey, MenuReducer, SetBadgeAction } from '@ansyn/menu';
+import { SetDropsAction } from '../../../overlays/actions/overlays.actions';
```
[the remark this answers is not in the document]
Comment (ramtob) [reply src/app/@ansyn/ansyn/app-effects/effects/filters.app.effects.spec.ts]: done

File: src/app/@ansyn/ansyn/app-effects/effects/filters.app.effects.ts [hunk 6]
```diff
@@ -42,39 +42,56 @@ import {
...
 	IFilterModel,
 	InjectionResolverFilter,
-	IOverlay,
+	IOverlay, IOverlaySpecialObject,
 	selectFavoriteOverlays,
 	selectRemovedOverlays,
 	selectRemovedOverlaysVisibility
 } from '@ansyn/core';
 import { filter, map, mergeMap, share, tap, withLatestFrom } from 'rxjs/operators';
+import { selectSpecialObjects } from '../../../overlays/reducers/overlays.reducer';
```
[the remark this answers is not in the document]
Comment (ramtob) [reply src/app/@ansyn/ansyn/app-effects/effects/filters.app.effects.ts]: not in this file anymore

File: src/app/@ansyn/core/utils/misc.ts [hunk 7]
```diff
@@ -0,0 +1,3 @@
+export function mapToArray(inMap: Map<any, any>): any[] {
```
[the remark this answers is not in the document]
Comment (ramtob) [reply src/app/@ansyn/core/utils/misc.ts]: done + more uses of mapValuesToArray()

File: src/app/@ansyn/core/utils/overlays.ts (hunk 1 above)
Comment (ramtob) [reply src/app/@ansyn/core/utils/overlays.ts]: 😎

File: src/app/@ansyn/menu-items/filters/services/filters.service.ts (hunk 2 above)
Comment (ramtob) [reply src/app/@ansyn/menu-items/filters/services/filters.service.ts]: done

File: src/app/@ansyn/overlays/effects/overlays.effects.ts (hunk 3 above)
Comment (ramtob) [reply src/app/@ansyn/overlays/effects/overlays.effects.ts]: 👍

File: src/app/@ansyn/ansyn/app-effects/effects/filters.app.effects.spec.ts [hunk 8]
```diff
@@ -34,7 +34,7 @@ import {
 	OverlaysService, overlaysStatusMessages, SetFilteredOverlaysAction, SetOverlaysStatusMessage
 } from '@ansyn/overlays';
 import { menuFeatureKey, MenuReducer, SetBadgeAction } from '@ansyn/menu';
-import { SetDropsAction } from '../../../overlays/actions/overlays.actions';
+import { SetDropsAction } from 'src/app/@ansyn/overlays/public_api';
```
Comment (yairtawil) [inline src/app/@ansyn/ansyn/app-effects/effects/filters.app.effects.spec.ts]: ansyn/overalys

File: src/app/@ansyn/ansyn/app-effects/effects/filters.app.effects.ts [hunk 9]
```diff
@@ -42,15 +43,15 @@ import {
...
+	IOverlaySpecialObject,
 	selectFavoriteOverlays,
 	selectRemovedOverlays,
 	selectRemovedOverlaysVisibility
 } from '@ansyn/core';
 import { filter, map, mergeMap, share, tap, withLatestFrom } from 'rxjs/operators';
-import { selectSpecialObjects } from '../../../overlays/reducers/overlays.reducer';
-import { SetDropsAction } from '../../../overlays/actions/overlays.actions';
-import { mapToArray } from '../../../core/utils/misc';
+import { selectSpecialObjects, SetDropsAction } from 'src/app/@ansyn/overlays/public_api';
```
Comment (yairtawil) [inline src/app/@ansyn/ansyn/app-effects/effects/filters.app.effects.ts]: 🔝

File: src/app/@ansyn/menu-items/filters/models/metadata/enum-filter-metadata.ts [hunk 10]
```diff
@@ -1,6 +1,7 @@
 import { FilterMetadata } from './filter-metadata.interface';
 import { IFilter } from '../IFilter';
 import { FilterType } from '@ansyn/core';
+import { mapValuesToArray } from 'src/app/@ansyn/core/public_api';
```
Comment (yairtawil) [inline src/app/@ansyn/menu-items/filters/models/metadata/enum-filter-metadata.ts]: 🔝  of course

File: src/app/@ansyn/overlays/reducers/overlays.reducer.ts [hunk 11]
```diff
@@ -3,7 +3,7 @@ import { LoadOverlaysSuccessAction, OverlaysActions, OverlaysActionTypes } from
 import { createFeatureSelector, createSelector, MemoizedSelector } from '@ngrx/store';
 import * as _ from 'lodash';
 import { ExtendMap } from './extendedMap.class';
-import { mapToArray } from '../../core/utils/misc';
+import { mapValuesToArray } from 'src/app/@ansyn/core/public_api';
```
Comment (yairtawil) [inline src/app/@ansyn/overlays/reducers/overlays.reducer.ts]: 🔝 🔢

File: src/app/@ansyn/overlays/services/overlays.service.ts [hunk 12]
```diff
@@ -5,7 +5,7 @@ import { IOverlay, IOverlaysCriteria, IOverlaysFetchData } from '@ansyn/core';
 import { IOverlayDropSources, ITimelineRange, OverlayDrop } from '../reducers/overlays.reducer';
 import { IOverlaysConfig } from '../models/overlays.config';
 import { unionBy } from 'lodash';
-import { mapToArray } from '../../core/utils/misc';
+import { mapValuesToArray } from 'src/app/@ansyn/core/public_api';
```
Comment (yairtawil) [inline src/app/@ansyn/overlays/services/overlays.service.ts]: :top thanks

File: src/app/@ansyn/ansyn/app-effects/effects/filters.app.effects.spec.ts (hunk 8 above)
Comment (ramtob) [reply src/app/@ansyn/ansyn/app-effects/effects/filters.app.effects.spec.ts]: drops

File: src/app/@ansyn/ansyn/app-effects/effects/filters.app.effects.ts (hunk 9 above)
Comment (ramtob) [reply src/app/@ansyn/ansyn/app-effects/effects/filters.app.effects.ts]: drops

Comment (ramtob) [reply src/app/@ansyn/ansyn/app-effects/effects/filters.app.effects.ts]: done

File: src/app/@ansyn/menu-items/filters/models/metadata/enum-filter-metadata.ts (hunk 10 above)
Comment (ramtob) [reply src/app/@ansyn/menu-items/filters/models/metadata/enum-filter-metadata.ts]: done

File: src/app/@ansyn/overlays/reducers/overlays.reducer.ts (hunk 11 above)
Comment (ramtob) [reply src/app/@ansyn/overlays/reducers/overlays.reducer.ts]: done

File: src/app/@ansyn/overlays/services/overlays.service.ts (hunk 12 above)
Comment (ramtob) [reply src/app/@ansyn/overlays/services/overlays.service.ts]: done thanks

**Pull request merged**